lib/matplotlib/image.py

```diff
@@ -610,6 +613,23 @@ def set_data(self, A):
                 or self._A.ndim == 3 and self._A.shape[-1] in [3, 4]):
             raise TypeError("Invalid dimensions for image data")
 
+        if self._A.ndim == 3:
+            # If the input data has values outside the valid range (after
+            # normalisation), we issue a warning and then clip X to the bounds
+            # - otherwise casting wraps extreme values, hiding outliers and
+            # making reliable interpretation impossible.
+            high = 255 if np.issubdtype(self._A.dtype, np.integer) else 1
+            if self._A.min() < 0 or high < self._A.max():
+                _log.warn(
```

Comment:
Its `_log.warning`, isn't it?

Reply:
Yep, turns out that `_log.warn` ["is functionally identical but deprecated"](https://docs.python.org/3/library/logging.html#logging.Logger.warning) (but not programmatically deprecated).